src/transformers/models/deta/modeling_deta.py

```diff
@@ -1953,6 +1967,7 @@ def forward(
                 aux_weight_dict = {}
                 for i in range(self.config.decoder_layers - 1):
                     aux_weight_dict.update({k + f"_{i}": v for k, v in weight_dict.items()})
+                aux_weight_dict.update({k + "_enc": v for k, v in weight_dict.items()})
```

Comment:
I realise this is just copying the code above - but copying the weight dict for all the layers is massively redundant if they're all the same.

Reply:
@NielsRogge This is different from just copying weight_dict, aux_weight also requires intermediate output loss for computation (e.g. loss_ce_enc and loss_ce_0 is different.)

Reply:
But all that's happening here is the weight dict is being copied and the keys are being changed. Likewise, on L1969, the dictionary is being copied `self.config.decoder_layers - 1` times, just with the keys being updated. So the weights for the respective losses is the same for all layers.

Reply:
@amyeroberts Well this is the code from original deformable DETR 
https://github.com/jozhang97/DETA/blob/f47e50efeb194357fb93a119f196a2e485fffdb5/models/deformable_detr.py#L549

https://github.com/huggingface/transformers/blob/1d7773594754457ed4a79cf6d98bcaabea5bff51/src/transformers/models/detr/modeling_detr.py#L1607C1-L1607C1

They all have same code.

Reply:
Since the original weight_dict will be like
```
 {"loss_ce": 1, "loss_bbox": 2, "loss_giou": 1}
```
and when the auxliary_loss is set true, it supposed to be changed like
```
 {
"loss_ce": 1, "loss_bbox": 2, "loss_giou": 1, # for original loss
"loss_ce_0": 1, "loss_bbox_0": 2, "loss_giou_0": 1, # for first layer output of decoder
...
"loss_ce_4": 1, "loss_bbox_4": 2, "loss_giou_4": 1, # for fourth layer output of decoder
"loss_ce_enc": 1, "loss_bbox_enc": 2, "loss_giou_enc": 1, # for encoder layer output
}

```

Reply:
Do you have any other recommendation to avoid simply duplicating the original keys?

Reply:
@amyeroberts 
```
weight_dict = {"loss_ce": 1, "loss_bbox": self.config.bbox_loss_coefficient}
weight_dict["loss_giou"] = self.config.giou_loss_coefficient

if self.config.auxiliary_loss:
    for i in range(self.config.decoder_layers - 1):
        for key in weight_dict.keys():
            weight_dict[key + f"_{i}"] = weight_dict[key]
    for key in weight_dict.keys():
        weight_dict[key + "_enc"] = weight_dict[key]

loss = sum(loss_dict[k] * weight_dict[k] for k in loss_dict.keys() if k in weight_dict)
```
Are you thinking something like this? Well personally I prefer to maintain the original way but this will works!

Reply:
My more general point is that we don't need to create this massive weight dict at all -> it could exist with just three keys as the weights for giou, bbox and ce are the same for all layers. 

Let's just leave as-is for now

Reply:
now i understood what you really meant, but let's just leave as-is for now because changing this might need some modification at the line 1972